sale_order_delivery_products/models/sale_order.py

```diff
@@ -0,0 +1,29 @@
+# Copyright Copyright 2021 Daniel Dominguez, Manuel Calero - https://xtendoo.es
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_view_picking(self):
+        pick_ids = self.mapped("picking_ids").sorted("id")
+        action = self.env.ref("stock.action_picking_tree_all")
+        result = action.read()[0]
+        # override the context to get id of the default filtering on operation type
+        result["context"] = {
+            "default_partner_id": self.partner_id.id,
+            "default_origin": self.name,
+        }
+        result["domain"] = [("id", "in", pick_ids.ids)]
+        result["res_id"] = pick_ids[:1].id
```

Comment:
don't you want to filter on the state of the pickings here? I can easily think of cases where the first picking is done, and this would defeat the purpose of the module.

Reply:
Are you proposing to open the first non-validated picking?